### PR TITLE
Add yajltcl add_latlon_list command

### PR DESCRIPTION
**Pull request by pm-conej**
* add the add_latlon_list command to yajltcl to speed up waypoint serialization

### CODE REVIEW

File: generic/yajltcl.c [hunk 1]
```diff
@@ -851,6 +853,36 @@ yajltcl_yajlObjectObjCmd(ClientData cData, Tcl_Interp *interp, int objc, Tcl_Obj
...
+            int lines;
+            char *latlon;
+            int len;
+
+            if (arg + 1 >= objc) {
+                Tcl_WrongNumArgs (interp, 1, objv, "list value");
+                return TCL_ERROR;
+            }
+
+            if (Tcl_ListObjGetElements(interp, objv[++arg], &lines, &linesPtr) == TCL_ERROR) {
```
Comment (bovine) [inline generic/yajltcl.c]: Maybe also validate the `lines` is an even number of elements?

**Comment by resuna** [comment]
Maybe a generalized add_tagged_list {lat lon} $lat_lon_list ?

**Comment by gahr** [comment]
I’ve had the same idea initially, in the form of {k1 k2 .. km} {v1 v2 .. vn} | m <= n, -> { “k1”: v1, “k2”: v2, .. “k(n%m)”: vn }, but then I realized that it’s unfortunately still not generic enough, as you requires values to be numeric...

File: generic/yajltcl.c (hunk 1 above)
Comment (pm-conej) [reply generic/yajltcl.c]: Added a check for even list.

**Comment by resuna** [comment]
The code here doesn't appear to be checking for numeric, so why does that need to be a constraint on the generalized code?

**Comment by gahr** [comment]
Because of `yajl_gen_number`

**Comment by resuna** [comment]
Too fast.

**Comment by bovine** [comment]
Not necessary any more due to new internal design direction.

**Pull request closed without merging**